inspirehep/records/marshmallow/literature/base.py

```diff
@@ -5,314 +5,46 @@
-
-    @post_dump
-    def post_dump(self, data):
-        data = strip_empty_values(data)
-        if is_superuser_or_cataloger_logged_in():
-            data["can_edit"] = True
-        return data
-
-
-class LiteratureESEnhancementV1(
-    ElasticSearchBaseSchema, LiteratureMetadataRawAdminSchemaV1
-):
-    """Elasticsearch serialzier"""
-
-    _ui_display = fields.Method("get_ui_display", dump_only=True)
-    _collections = fields.Raw(dump_only=True)
-    abstracts = fields.Nested(AbstractSource, dump_only=True, many=True)
-    author_count = fields.Method("get_author_count")
-    authors = fields.Nested(AuthorsInfoSchemaForES, dump_only=True, many=True)
-    bookautocomplete = fields.Method("get_bookautocomplete")
-    earliest_date = fields.Raw(dump_only=True, default=missing)
-    facet_inspire_doc_type = fields.Method("get_inspire_document_type")
-    facet_author_name = fields.Method("get_facet_author_name")
-    id_field = fields.Integer(dump_only=True, dump_to="id", attribute="control_number")
-    thesis_info = fields.Nested(ThesisInfoSchemaForESV1, dump_only=True)
-
-    def get_ui_display(self, record):
-        return json.dumps(LiteratureMetadataUISchemaV1().dump(record).data)
-
-    def get_author_count(self, record):
-        """Prepares record for ``author_count`` field."""
-        authors = record.get("authors", [])
-
-        authors_excluding_supervisors = [
-            author
-            for author in authors
-            if "supervisor" not in author.get("inspire_roles", [])
-        ]
-        return len(authors_excluding_supervisors)
-
-    def get_inspire_document_type(self, record):
-        """Prepare record for ``facet_inspire_doc_type`` field."""
-        result = []
-
-        result.extend(record.get("document_type", []))
-        result.extend(record.get("publication_type", []))
-        if "refereed" in record and record["refereed"]:
-            result.append("peer reviewed")
-        return result
-
-    def get_facet_author_name(self, record):
-        """Prepare record for ``facet_author_name`` field."""
-        authors_with_record = list(
-            InspireRecord.get_linked_records_from_dict_field(record, "authors.record")
-        )
-        found_authors_control_numbers = set(
-            [
-                author["control_number"]
-                for author in authors_with_record
-                if author.get("control_number")
-            ]
-        )
-        authors_without_record = [
-            author
-            for author in record.get("authors", [])
-            if "record" not in author
-            or int(
-                PidStoreBase.get_pid_from_record_uri(author["record"].get("$ref"))[1]
-            )
-            not in found_authors_control_numbers
+        exclude = FIELDS_TO_EXCLUDE + [
+            "_collections",
+            "_desy_bookkeeping",
+            "_export_to",
+            "_files",
+            "_private_notes",
+            "self",
```

Comment:
should ``self`` excluded by other collections as well or it's just a literature thing?

Reply:
not really important, but we can exclude everywhere except indexing